src/Switch-MBRAUN_SCU101/main.py

```diff
@@ -0,0 +1,220 @@
+
+    def measure(self) -> None:
+        """Measure the current state of the shutter."""
+        self.shutter_state = self.get_shutter_state()
+
+        if self.shutter_state_dict[self.shutter_state] == "Not connected":
+            msg = "Unable to move the shutter as the controller is unable to connect to it."
+            raise Exception(msg)
+        elif self.shutter_state_dict[self.shutter_state] == "Blocked":
+            msg = "Unable to move the shutter as it is blocked."
+            raise Exception(msg)
+
+    def call(self) -> str:
+        """Return the current state of the device."""
+        return self.shutter_state_dict[self.shutter_state]
+
+    """ SCU101 specific functions"""
+
+    def get_shutter_state(self) -> int:
+        """Read the current state of the shutter."""
+        register_address = 0x0000 if self.shutter_number == 1 else 0x0020
+        mask = 0x000F
+
+        return self.read_register(register_address, mask)
+
+    def read_register(self, address: int, mask: hex) -> int:
+        """Read a register from the device."""
+        response = self.modbus.read_register(address, 0, functioncode=4)
+        return response & mask
+
+    def handle_set_state_input(self, set_state_input: float | int | bool) -> hex:
+        """Handle the input for the state of the shutter and return hex representation."""
+        if isinstance(set_state_input, (float, int)):
+            state_hex = 0x0001 if int(set_state_input) == 1 else 0x0002
+
+        elif isinstance(set_state_input, bool):
+            state_hex = 0x0001 if set_state_input else 0x0002
+
+        elif isinstance(set_state_input, str):
+            state_hex = 0x0001 if set_state_input.lower() == "open" else 0x0002
+
+        else:
+            msg = "Input of %s cannot be transformed to allowed shutter states of 0x0001 or 0x0002." % input
+            raise Exception(msg)
+
+        return state_hex
+
+    def set_shutter_state(self, target_shutter_state: int) -> None:
+        """Set the state of the shutter."""
+        register_address = 0x0001 if self.shutter_number == 1 else 0x0021
+        shutter_set_mask = 0x0003
+        execute_flag_mask = 0x0004
+
+        # Set the new shutter position
+        current_value = self.read_register(register_address, shutter_set_mask)
+
+        # Masks for setting and clearing bits
+        clear_mask = ~(shutter_set_mask | execute_flag_mask)  # Combine masks and invert to create a clear mask
+        # Isolate bits for the new shutter position
+        set_shutter_position_mask = target_shutter_state & shutter_set_mask
+
+        # Apply masks to current_value
+        # First, clear the bits related to the shutter set and execute flag in the current value
+        cleared_current_value = current_value & clear_mask
+        # Then, set the shutter position bits in the cleared current value
+        value_to_write = cleared_current_value | set_shutter_position_mask
```

Comment:
I think the whole bitwise mask handling is correct, but it looks really complicated.
In the end it's basically just a kind of set_new_value_with_mask?
like 
```python
def set_new_value_with_mask(old_value, new_value, mask):
    return (old_value &~ mask) | (new_value & mask)
```

Reply:
Maybe it's useful to at least add a comment that the point of the bit handling is to set the "set" and "execute" bits of the shutter while retained all other bits of the state